block/meta-iosched.c

```diff
@@ -288,6 +320,10 @@ void update_stats(struct request_queue *q) {
 	whole_stat++;
 	spin_unlock(&stat_lock);
 	q->stats[group]++;
+
+	//spin_lock(q->queue_lock); //spin_lock_irq(q->queue_lock);
```

Comment:
AFAICS, you call this from dispatch function, which is called with q->queue_lock already being held, so you are trying to acquire the same lock twice - deadlock.